pandas/io/pytables.py

```diff
@@ -916,7 +917,8 @@ def select_as_multiple(
                 raise ValueError("all tables must have exactly the same nrows!")
 
         # axis is the concentration axes
-        axis = list({t.non_index_axes[0][0] for t in tbls})[0]
+        _tbls = cast(List[Table], tbls)  # assured by check above
```

Comment:
from the contributing guide `.. every effort should be exhausted to avoid ``cast```

the cast could maybe be avoided with something like
 
```python
        def _validate_tables(tbls: List[Any]) -> List[Table]:
            for tbl in tbls:
                if not isinstance(tbl, Table):
                    raise TypeError(
                        "object [{obj}] is not a table, and cannot be used in all "
                        "select as multiple".format(obj=t.pathname)
                    )
            return tbls

        _tbls = _validate_tables(tbls)
```

Reply:
how about replacing the cast here with `_tbls = [x for x in tbls if isinstance(x, Table)]` along with a comment that the check is redundant?

Reply:
good idea.